pkg/admission/prep_downscale.go

```diff
@@ -329,3 +338,35 @@ func findDownscalesDoneMinTimeAgo(ctx context.Context, api kubernetes.Interface,
 	}
 	return nil, nil
 }
+
+// findStatefulSetWithNonUpdatedReplicas returns any statefulset that has non-updated replicas, indicating that the statefulset
+// may be in the process of being rolled.
+func findStatefulSetWithNonUpdatedReplicas(ctx context.Context, api kubernetes.Interface, namespace, rolloutGroup string) (*statefulSetDownscale, error) {
+	stsList, err := findStatefulSetsForRolloutGroup(ctx, api, namespace, rolloutGroup)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, sts := range stsList.Items {
+		status := sts.Status
+		if !(status.Replicas == status.ReadyReplicas && status.ReadyReplicas == status.UpdatedReplicas) {
+			return &statefulSetDownscale{
+				name:               sts.Name,
+				nonUpdatedReplicas: uint(status.Replicas - status.UpdatedReplicas),
```

Comment:
Seems like this would be deceptive if there are not ready replicas. Should we return both non-ready and non-updated here?

Reply:
Good suggestion